### PR TITLE
Avoid adding redundant `-c` and `-o` flags when compiling. NFC.

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -2660,6 +2629,16 @@ def consume_arg_file():
       options.shared = True
     elif check_flag('-r'):
       options.relocatable = True
+    elif check_arg('-o'):
+      options.output_file = consume_arg()
+    elif arg.startswith('-o'):
+      options.output_file = arg[2:]
+      newargs[i] = ''
+    elif arg == '-mllvm':
+      # Ignore the next argument which rather than trying to parse it.
```
Comment (tlively) [inline emcc.py:2638]:
```suggestion
      # Ignore the next argument rather than trying to parse it.
```

File: emcc.py [hunk 2]
```diff
@@ -2064,7 +2026,9 @@ def compile_source_file(i, input_file):
         cmd = get_clang_command_asm(input_file)
       else:
         cmd = get_clang_command(input_file)
-      cmd += ['-c', '-o', output_file]
+      if not compile_only:
+        cmd += ['-c']
```
Comment (tlively) [inline emcc.py:2030]: This seems counterintuitive. I would have expected `-c` to be used when we only want to compile. Why is the condition `not compile_only`? Also, where does `compile_only` come from?

File: emcc.py [hunk 3]
```diff
@@ -2660,6 +2629,16 @@ def consume_arg_file():
...
       options.relocatable = True
+    elif check_arg('-o'):
+      options.output_file = consume_arg()
+    elif arg.startswith('-o'):
+      options.output_file = arg[2:]
+      newargs[i] = ''
+    elif arg == '-mllvm':
+      # Ignore the next argument which rather than trying to parse it.
+      # This is needed because llvm args could, for example, start with `-o` and
+      # we don't won't to confure that with a normal `-o` flag.
```
Comment (tlively) [inline emcc.py:2640]:
```suggestion
      # we don't want to confuse that with a normal `-o` flag.
```

File: emcc.py (hunk 2 above)
Comment (sbc100) [reply emcc.py:2030]: Agree its confusing.   `compile_only` basically means that that user specified `-c` themselves so we don't need to add it:

```compile_only = has_dash_c or has_dash_S or has_dash_E```

I changed to `if not has_dash_c:` which is more correct and more readable.

**Review by tlively — APPROVED** [review]

**Pull request merged**